front/src/applications/operationalStudies/views/v2/ScenarioV2.tsx

```diff
@@ -18,179 +17,91 @@ import type {
-  } = osrdEditoastApi.endpoints.getV2ProjectsByProjectIdStudiesAndStudyIdScenariosScenarioId.useQuery(
-    {
-      projectId: projectId!,
-      studyId: studyId!,
-      scenarioId: scenarioId!,
-    },
-    {
-      skip: !projectId || !studyId || !scenarioId,
-    }
-  );
-
-  useEffect(() => {
-    if (isScenarioError && errorScenario) throw errorScenario;
-  }, [isScenarioError, errorScenario]);
-
-  const { updateInfraID, updateTimetableID, updateElectricalProfileSetId } = useOsrdConfActions();
-
-  useEffect(() => {
-    if (scenario) {
-      dispatch(updateTimetableID(scenario.timetable_id));
-      dispatch(updateInfraID(scenario.infra_id));
-      dispatch(updateElectricalProfileSetId(scenario.electrical_profile_set_id));
-    }
-  }, [scenario]);
-
-  const { data: timetable } = osrdEditoastApi.endpoints.getV2TimetableById.useQuery(
-    { id: timetableId! },
-    {
-      skip: !timetableId,
-    }
-  );
-
-  const { data: conflicts } = osrdEditoastApi.endpoints.getV2TimetableByIdConflicts.useQuery(
-    {
-      id: scenario?.timetable_id as number,
-      infraId: scenario?.infra_id as number,
-      electricalProfileSetId: scenario?.electrical_profile_set_id,
-    },
-    { skip: !scenario }
-  );
-
-  useEffect(() => {
-    if (timetable && infra?.state === 'CACHED' && timetable.train_ids.length > 0) {
-      selectProjectionV2(timetable.train_ids, trainIdUsedForProjection, selectedTrainId);
-    }
-  }, [timetable, infra]);
-
-  useEffect(() => {
-    if (timetable && infra?.state === 'CACHED' && trainIdUsedForProjection && infraId) {
-      // If trainResultsToFetch is undefined that means it's the first load of the scenario
-      // and we want to get all timetable trains results
-      getSpaceTimeChartData(
-        trainResultsToFetch ?? timetable.train_ids,
-        trainIdUsedForProjection,
-        infraId,
-        setTrainSpaceTimeData,
-        scenario?.electrical_profile_set_id ?? undefined
-      );
-    }
-  }, [timetable, trainIdUsedForProjection, infra]);
-
-  useEffect(() => {
-    if (!projectId || !studyId || !scenarioId) {
-      throw new Error('Missing projectId, studyId or scenarioId');
-    }
-  }, [projectId, studyId, scenarioId]);
-
-  useEffect(
-    () => () => {
-      dispatch(updateTimetableID(undefined));
-      dispatch(updateInfraID(undefined));
-      dispatch(updateElectricalProfileSetId(undefined));
-      dispatch(updateTrainIdUsedForProjection(undefined));
-    },
-    []
-  );
+  const scenarioData = useScenarioData();
 
   const toggleMicroMacroButton = (isMacroMode: boolean) => {
     setIsMacro(isMacroMode);
     setCollapsedTimetable(isMacroMode);
   };
 
-  const handleNGEOperation = (event: NGETrainrunEvent, netzgrafikDto: NetzgrafikDto) =>
-    handleOperation({ event, dispatch, timeTableId: timetableId!, netzgrafikDto });
+  const handleNGEOperation = (event: NGETrainrunEvent, netzgrafikDto: NetzgrafikDto) => {
+    if (scenarioData?.scenario.timetable_id) {
```

Comment:
I believe we can just use `scenarioData!` here. If we don't have a scenario, then NGE shouldn't be active at all. (I'd prefer to notice any such unforeseen cases with a hard error, instead of silently dropping/ignoring the NGE events.)

Reply:
Done :+1: